gooddata-sdk/gooddata_sdk/catalog/declarative/analytics_layer.py

```diff
@@ -0,0 +1,167 @@
+            analytical_dashboards = [
+                CatalogDeclarativeAnalyticalDashboard.from_api(v) for v in entity["analytical_dashboards"]
+            ]
+        if entity.get("dashboard_plugins"):
+            dashboard_plugins = [CatalogDeclarativeDashboardPlugin.from_api(v) for v in entity["dashboard_plugins"]]
+        if entity.get("filter_contexts"):
+            filter_contexts = [CatalogDeclarativeFilterContext.from_api(v) for v in entity["filter_contexts"]]
+        if entity.get("metrics"):
+            metrics = [CatalogDeclarativeMetric.from_api(v) for v in entity["metrics"]]
+        if entity.get("visualization_objects"):
+            visualization_objects = [
+                CatalogDeclarativeVisualizationObject.from_api(v) for v in entity["visualization_objects"]
+            ]
+
+        return cls(analytical_dashboards, dashboard_plugins, filter_contexts, metrics, visualization_objects)
+
+    def to_api(self) -> DeclarativeAnalyticsLayer:
+        kwargs: dict[str, Any] = {
+            "analytical_dashboards": [],
+            "dashboard_plugins": [],
+            "filter_contexts": [],
+            "metrics": [],
+            "visualization_objects": [],
+        }
+
+        if self.analytical_dashboards:
+            kwargs["analytical_dashboards"] = [v.to_api() for v in self.analytical_dashboards]
+        if self.dashboard_plugins:
+            kwargs["dashboard_plugins"] = [v.to_api() for v in self.dashboard_plugins]
+        if self.filter_contexts:
+            kwargs["filter_contexts"] = [v.to_api() for v in self.filter_contexts]
+        if self.metrics:
+            kwargs["metrics"] = [v.to_api() for v in self.metrics]
+        if self.visualization_objects:
+            kwargs["visualization_objects"] = [v.to_api() for v in self.visualization_objects]
+        return DeclarativeAnalyticsLayer(**kwargs)
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, CatalogDeclarativeAnalyticsLayer):
+            return False
+        return (
+            self.analytical_dashboards == other.analytical_dashboards
+            and self.dashboard_plugins == other.dashboard_plugins
+            and self.filter_contexts == other.filter_contexts
+            and self.metrics == other.metrics
+            and self.visualization_objects == other.visualization_objects
+        )
+
+
+class CatalogAnalyticsBase:
+    def __init__(self, id: str, title: str, content: dict[str, Any], description: str = None, tags: list[str] = None):
+        self.id = id
+        self.title = title
+        self.content = content
+        self.description = description
+        self.tags = tags
+
+    @classmethod
+    def from_api(cls: Type[T], entity: dict[str, Any]) -> T:
+        return cls(entity["id"], entity["title"], entity["content"], entity.get("description"), entity.get("tags"))
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, CatalogAnalyticsBase):
```

Comment:
This together with fact `__eq__` is not overloaded in descendants allows to compare descendants between each other. May be, the following should be used:
```
if not isinstance(other, self.__class__):
```
Note, I have not tested above suggestion. There can be hidden devil in it.

Reply:
Thanks. Tested. Works.

from gooddata_sdk.catalog.declarative.analytics_layer import *
m = CatalogDeclarativeMetric("a", "a", {"a":"a"}, "a",["a"])
o = CatalogDeclarativeVisualizationObject("a", "a", {"a":"a"}, "a",["a"])
m == o # False